mne/viz/_brain/_brain.py

```diff
@@ -474,6 +474,10 @@ def add_data(self, array, fmin=None, fmid=None, fmax=None,
                 actor, mesh = mesh_data
                 # add metadata to the mesh for picking
                 mesh._hemi = hemi
+                mapper = actor.GetMapper()
+                mapper.SetResolveCoincidentTopologyToPolygonOffset()
+                mapper.SetRelativeCoincidentTopologyPolygonOffsetParameters(
+                    -1., -1.)
```

Comment:
What I would do here is make it implicitly that the order of addition sets the zorder (things plotted later end up on top). So `brain` should have something like a `self._polygon_offset_counter = 0` in init and anything you add afterward does `self._polygon_offset_counter += 1` and then `mapper.SetRelativeCoincidentTopologyPolygonOffsetParameters(-self._polygon_offset_counter, -self._polygon_offset_counter)` or so. WDYT?

Reply:
From my experience with polygon offset, especially if many surfaces overlap, some artifacts can appear. Also, I checked the doc of SetRelativeCoincidentTopologyPolygonOffsetParameter and it accepts double so maybe we have some room for tweaks here.

Ref: https://vtk.org/doc/nightly/html/classvtkMapper.html#ad49f0d5f3086b3a79f542cb9e801d8ac

Reply:
I'll go with `self._polygon_offset_counter` and modify accordingly if we run into artifacts :+1:

Reply:
It seems like as soon as we do a `brain.add_annotation` plus a `brain.add_data` we will have a problem, but I guess we can deal with it then